editor.planx.uk/src/pages/FlowEditor/components/Settings/GeneralSettings/BoundaryForm.tsx

```diff
@@ -21,12 +29,23 @@ export default function BoundaryForm({ formikConfig, onSuccess }: FormProps) {
     ...formikConfig,
     validationSchema: formSchema,
     onSubmit: async (values, { resetForm }) => {
-      const isSuccess = await useStore.getState().updateTeamSettings({
-        boundaryUrl: values.boundaryUrl,
-      });
-      if (isSuccess) {
-        onSuccess();
-        resetForm({ values });
+      try {
+        const { data } = await axios.get(`${values.boundaryUrl}.geojson`);
+        const bboxPoly = bboxPolygon(bbox(data));
+        const bboxFeature = feature(bboxPoly.geometry);
+
+        const isUpdateSuccess = await useStore.getState().updateTeamSettings({
+          boundaryUrl: values.boundaryUrl,
+          boundaryBbox: bboxFeature,
+        });
+        if (isUpdateSuccess) {
+          onSuccess();
+          resetForm({ values });
+        }
+      } catch (error) {
+        formik.errors.boundaryUrl =
+          "We are unable to retrieve your boundary, check your boundary URL and try again";
+        console.error(error);
```

Comment:
This looks good!

We might want slightly more fine-grained checking of the error type (i.e. using `isAxiosError()`) or a bit more control over the URL we send the request to.

Reply:
Addressed here - [7c1d9fb](https://github.com/theopensystemslab/planx-new/pull/3378/commits/7c1d9fb8ba8a738e661c2733ba49d70fb3af0589)

I've also used the helper method `setFieldError()` as there may be a side effect of directly modifying the `errors` object ([docs](https://formik.org/docs/api/formik#setfielderror-field-string-errormsg-string--void)).